### PR TITLE
Fix cmpStrings in js mode

### PR DESCRIPTION
**Pull request by treeform**
Fixing: https://github.com/nim-lang/Nim/issues/7603

### CODE REVIEW

**Comment by Araq** [comment]
Will merge once the tests are green, it's correct.

File: lib/system/jssys.nim [hunk 1]
```diff
@@ -414,14 +414,13 @@ proc SetMinus(a, b: int): int {.compilerproc, asmNoStackFrame.} =
 
 proc cmpStrings(a, b: string): int {.asmNoStackFrame, compilerProc.} =
   asm """
-    if (`a` == `b`) return 0;
```
Comment (yglukhov) [inline lib/system/jssys.nim]: This will not work for nil strings:
```nim
var a, b: string
echo cmp(a, b) # prints -1
```

Comment (treeform) [reply lib/system/jssys.nim]: You are right! Fixed.

**Comment by treeform** [comment]
Also added tests.

**Pull request merged**